client/web/src/enterprise/batches/preview/list/backend.ts

```diff
@@ -325,3 +327,33 @@ export const queryChangesetSpecFileDiffs = ({
             return node.description.diff.fileDiffs
         })
     )
+
+export const queryPublishableChangesetSpecIDs = ({
+    batchSpec,
+    search,
+    currentState,
+    action,
+}: BatchSpecApplyPreviewVariables): Observable<Scalars['ID'][]> => {
+    const request = (after: string | null): Observable<BatchSpecApplyPreviewConnectionFields> =>
+        queryChangesetApplyPreview({
+            batchSpec,
+            first: 10000,
+            after,
+            search,
+            currentState,
+            action,
+        })
+
+    return request(null).pipe(
+        expand(connection => (connection.pageInfo.hasNextPage ? request(connection.pageInfo.endCursor) : EMPTY)),
+        reduce(
+            (previous, next) =>
+                previous.concat(
+                    next.nodes
+                        .map(node => getPublishableChangesetSpecID(node))
+                        .filter((maybeID): maybeID is string => maybeID !== null)
```

Comment:
```suggestion
                        .filter((maybeID): maybeID is Scalars['ID'] => maybeID !== null)
```

_should_ enable you to remove the cast below :)